src/js/common/array.js

```diff
@@ -221,45 +224,155 @@ function stringDESCIgnoreCase(_a, _b) {
+ * @returns {number} Result of comparison.
+ */
+function _getAllDayCompare(a, b) {
+    return booleanASC(a.valueOf().isAllDay || a.hasMultiDates, b.valueOf().isAllDay || b.hasMultiDates);
+}
+
+/**
+ * Calculate start time, including the going duration of travel time
+ *
+ * @param {Schedule|ScheduleViewModel} a The object schedule instance.
+ * @param {Schedule|ScheduleViewModel} b The object schedule instance.
+ * @returns {number} Result of comparison.
+ */
+function _getStartCompare(a, b) {
+    var aTravelTime = getMaxTravelTime(a);
+    var bTravelTime = getMaxTravelTime(b);
+    var aGoingDuration = datetime.millisecondsFrom('minutes', aTravelTime.maxGoingDuration);
+    var bGoingDuration = datetime.millisecondsFrom('minutes', bTravelTime.maxGoingDuration);
+    var renderedStartA = new TZDate(a.getStarts().getTime() - aGoingDuration);
+    var renderedStartB = new TZDate(b.getStarts().getTime() - bGoingDuration);
+
+    return datetime.compare(renderedStartA, renderedStartB);
+}
+
+/**
+ * Compare time during start-end of schedule
+ *
+ * @param {Schedule|ScheduleViewModel} a The object schedule instance.
+ * @param {Schedule|ScheduleViewModel} b The object schedule instance.
+ * @returns {number} Result of comparison.
+ */
+function _getDurationCompare(a, b) {
+    var durationA = a.duration();
+    var durationB = b.duration();
 
     if (durationA < durationB) {
         return 1;
     }
+
     if (durationA > durationB) {
         return -1;
     }
 
-    return util.stamp(modelA) - util.stamp(modelB);
+    return 0;
+}
+
+/**
+ * Sort elements in duplicate arrays sequentially
+ *
+ * @param {Schedule|ScheduleViewModel} array The scheduleViewModel instance.
+ * @returns {Schedule|ScheduleViewModel[]} Sorted array
+ */
+function sortByDuplicate(array) {
+    var clone = array.slice(0);
+    var newArr = [];
+    var equal = [];
+    var el;
+
+    while (clone.length) {
+        el = clone[0];
+
+        newArr.push(el);
+        equal = _duplicates(clone, _isEqualVM(el));
+
+        if (equal.length) {
+            newArr = newArr.concat(equal);
+        }
+
+        clone = _remove(clone, _isEqualVM(el));
+    }
+
+    return newArr;
+}
+
+/**
+ * Returns a callback function that compares the viewModel's ID with the input model's ID.
+ *
+ * @param {Schedule|ScheduleViewModel} vm The scheduleViewModel instance.
+ * @returns {function} Callback function
+ */
+function _isEqualVM(vm) {
+    return function(value) {
+        return vm.model.id === value.model.id;
+    };
+}
+
+/**
+ * Filter duplicate elements in arrays except the first
+ *
+ * @param {Schedule|ScheduleViewModel[]} array The List of ScheduleViewModel(or Schedule)
+ * @param {function} condition Callback function to check filtering condition
+ * @returns {Schedule|ScheduleViewModel[]} Result of filtered
+ */
+function _duplicates(array, condition) {
+    return array.slice(1, array.length).filter(condition);
```

Comment:
The second parameter of `slice` is not necessary..

Reply:
Right.
You do not have to specify the second parameter.

> Zero-based index before which to end extraction. `slice` extracts up to but not including *end*.